src/AggregateFunctions/AggregateFunctionMinMaxAny.h

```diff
@@ -488,17 +485,61 @@ struct SingleValueDataString //-V730
 
     void insertResultInto(IColumn & to) const
     {
+        /// Same comment from write() applies here, but in this case we remove the last NULL character as we expect it to come from
+        /// old statuses that always included it
         if (has())
-            assert_cast<ColumnString &>(to).insertData(getData(), size);
-        else
-            assert_cast<ColumnString &>(to).insertDefault();
+        {
+            const char * data = getData();
+            size_t final_size = data[size - 1] != '\0' ? size : size - 1;
+            if (final_size)
+            {
+                assert_cast<ColumnString &>(to).insertData(data, final_size);
+                return;
+            }
+        }
+
+        assert_cast<ColumnString &>(to).insertDefault();
     }
 
     void write(WriteBuffer & buf, const ISerialization & /*serialization*/) const
     {
-        writeBinary(size, buf);
-        if (has())
-            buf.write(getData(), size);
+        /// This used to be serialized using insertDataWithTerminatingZero up until 22.9 when it was changed to use insertData
+        /// The problem is that this change, inadvertently, changed the serialization of the state, as insertDataWithTerminatingZero
+        /// included the final NULL in its size count, while insertData doesn't. That is a string like "1234" would have been inserted
+        /// as { size: 5, data: "1234\0" }, but with the change it was inserted as { size: 4, data: "1234\0" }-
```

Comment:
Seems like it's not correct, see https://github.com/ClickHouse/ClickHouse/issues/42916#issuecomment-1317425838
If I'm not mistaken, with the change it was inserted as `{ size: 4, data: "1234" }`

Reply:
`insertData` always adds a NULL:

https://github.com/ClickHouse/ClickHouse/blob/451fb07115e0740c4f28fb51b4891db3c94e3cfc/src/Columns/ColumnString.h#L160

Reply:
Ok, now I understand that you mean the binary representation and not the string representation. The string one always adds a NULL, but the binary doesn't. In the old version it got it from the string and with the change it stop getting it, so yes your are right.